### ISSUE TITLE
Null ref: T2.CreatureReviveTrigger.CreaturePerformingRevive_RemoveCreatureState

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#892]
Auto report. 0.2.6a. Feedback ID: 20210410-DQKF6. Couch co-op. Game over screen.

Likely need to use the newer `.Exists` check with null checking (or just add null checking in general).

```
6:48:39, Frame 47940, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.CreatureReviveTrigger.CreaturePerformingRevive_RemoveCreatureState ()
T2.CreatureReviveTrigger.DisableAllTheThings (System.Boolean tellServerAboutPrematureExit)
T2.CreatureReviveTrigger.Update ()

GameStates: RompingThroughDungeon, GameOverScreen
OnlineGameInfo: Device is offline.
No previous server nor online game info.
GameSetupMode: CouchCoop
Players: 3
DungeonRoomTransitionerState: Null
Room: 6 / 6
RoomModifiers: ScythesOnly (镰刀派对)
YouHaveDiedState: InitialDelay
SystemPlayer InputDevice: XInput Controller
HumanPlayer 1 InputDevice: XInput Controller
HumanPlayer 2 InputDevice: 键盘
HumanPlayer 3 InputDevice: XInput Controller
```

**Comment by ChrisNZL** [comment]
Another auto report. 0.2.8a. Feedback ID: 20210910-CJZWK

Slightly different stack trace.

```
8:52:34, Frame 77236, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.CreatureReviveTrigger.CreaturePerformingRevive_RemoveCreatureState ()
T2.CreatureReviveTrigger.DisableAllTheThings (System.Boolean tellServerAboutPrematureExit)
T2.CreatureReviveTrigger.OnCreatureStoodUp ()
T2.Creature.StandUp (T2.PlayerStruct playerStructWhoPerformedRevival)
T2.CreatureReviveTrigger.Update ()

GameStates: RompingThroughDungeon, GameOverScreen
OnlineGameInfo: Device is offline.
No previous server nor online game info.
GameSetupMode: CouchCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 12 / 12
RoomModifiers: Heartless (Heartless), BowsOnly (Archery Practice)
YouHaveDiedState: InitialDelay
SystemPlayer InputDevice: XInput Controller
HumanPlayer 1 InputDevice: Keyboard
HumanPlayer 2 InputDevice: XInput Controller
```

**Comment by ChrisNZL** [comment]
Further auto report, v0.3.0a, online, feedback ID 20220104-Y7GLD

```
9:54:18, Frame 282934, LOG »  Player 流狸 has fallen!
9:54:29, Frame 283590, LOG »  Player 宇宙级小甜甜子夜 has fallen!

9:54:29, Frame 283590, LOG »  GameStruct.OnAllPlayersDied: All players have died.
Party died in room number: 22
Highest room number reached: 22

9:54:29, Frame 283590, LOG »  YouHaveDied: All player creatures have died.


9:54:29, Frame 283590, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.CreatureReviveTrigger.CreaturePerformingRevive_RemoveCreatureState ()
T2.CreatureReviveTrigger.DisableAllTheThings (System.Boolean tellServerAboutPrematureExit)
T2.CreatureReviveTrigger.StopPrematurelyFromNetworkMessage ()
T2.NetworkMessage_StartRevivingCreature.ProcessReceivedPacket (T2.NetworkPeer recipient, LiteNetLib.NetPeer sender, LiteNetLib.Utils.NetDataReader reader, T2.GameStruct gameStruct)
T2.NetworkPeer.OnNetworkReceive (LiteNetLib.NetPeer peerThatSentMessage, LiteNetLib.NetPacketReader reader, LiteNetLib.DeliveryMethod deliveryMethod)
LiteNetLib.NetManager.ProcessEvent (LiteNetLib.NetEvent evt)
LiteNetLib.NetManager.PollEvents ()
T2.NetworkClient.Update ()

GameStates: RompingThroughDungeon, GameOverScreen
OnlineGameInfo: Device is online.
Server: Lightning-Beijing-01
GameID: HBHWB
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 22 / 22
RoomModifiers: LargeRoom (大型房间), LightningOrbsOnly (全面通电), ExtraChests (更多宝箱), Tilted (歪歪斜斜)
YouHaveDiedState: InitialDelay
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.3.1. Added some null checks within the `CreatureReviveTrigger` class.
